cleaner/users/classes/clean.php

```diff
@@ -96,13 +98,39 @@ private static function replace_usernames() {
         $DB->execute($sql);
     }
 
+    private static function replace_first_and_last_names() {
+        global $DB;
+
+        echo "Updating all first and last names...\n";
+
+        $where = 'id IN ('.self::$idstoupdate.')';
+        $firstprefix = self::FIRST_NAME_PREFIX;
+        $lastprefix = self::LAST_NAME_PREFIX;
+        $sql = <<<SQL
+UPDATE {user}
+SET firstname = CONCAT('{$firstprefix}', id), lastname = CONCAT('{$lastprefix}', id)
+WHERE $where
+SQL;
+        $DB->execute($sql);
+    }
+
     private static function scramble_fields() {
-        $fieldset = [
-            'main names'  => ['firstname', 'lastname'],
-            'other names' => ['firstnamephonetic', 'alternatename', 'middlename', 'lastnamephonetic'],
-            'department'  => ['institution', 'department'],
-            'address'     => ['address', 'city', 'country', 'lang', 'calendartype', 'timezone'],
-        ];
+        $config = get_config('cleaner_users');
+        if (isset($config->renameusers) && $config->renameusers == 1) {
+            $fieldset = [
+                'other names' => ['firstnamephonetic', 'alternatename', 'middlename', 'lastnamephonetic'],
+                'department'  => ['institution', 'department'],
+                'address'     => ['address', 'city', 'country', 'lang', 'calendartype', 'timezone'],
+            ];
+            self::replace_first_and_last_names();
```

Comment:
Should this be all or nothing? ie we either scramble all the names, OR we replace all the names. We don't want to do both? The scrambling is mildly expensive so we don't want to replace them and then needlessly scramble them again

Some unit tests would be good too